geopandas/geodataframe.py

```diff
@@ -1825,12 +1825,16 @@ def explode(self, column=None, ignore_index=False, index_parts=None, **kwargs):
 
         exploded_geom = self.geometry.reset_index(drop=True).explode(index_parts=True)
 
-        df = GeoDataFrame(
-            self.drop(self._geometry_column_name, axis=1).take(
-                exploded_geom.index.droplevel(-1)
-            ),
-            geometry=exploded_geom.values,
-        ).__finalize__(self)
+        df = (
+            GeoDataFrame(
+                self.drop(self._geometry_column_name, axis=1).take(
+                    exploded_geom.index.droplevel(-1)
+                ),
+                geometry=exploded_geom.values,
+            )
+            .rename(columns={"geometry": self._geometry_column_name})  # GH2501
```

Comment:
One corner case that I am thinking that won't work with this (but also not with the current code, but it might have worked before the changes of 0.11): if you have another non-active geometry dtype column that is actually called "geometry" ..

Not fully sure if we want to address that here (this should also be a fine regression fix to already merge), but I am thinking of doing something like:

```
df = self.drop(...)
df[exploded_geom.name] = exploded_geom.values  # or alternatively setting the index of `exploded_geom` to the same as `df`, and then doing a concat
df = df.set_geometry(self._geometry_column_name})
```

Reply:
or if we want to keep it with a single method chain: `df = self.drop(...).assign(**dict(exploded_geom.name=...).set_geometry(..)`

Reply:
Ah, good catch! I've used the suggestion above and added a test for this. We also no longer have `__finalize__` there but I guess that should not be an issue with this setup.